v2/internal/testcommon/samples_tester.go

```diff
@@ -0,0 +1,226 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1beta20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+const RefsPackage = "refs"
+const samplesPath = "../../config/samples"
+
+// exclusions slice contains resources to exclude from test as these resources run into
+// "runtime.notRegisteredErr" error from the Scheme.
+var exclusions = [...]string{"batchaccountpool", "snapshot",
+	"sqldatabasecontainerthroughputsetting", "sqldatabasethroughputsetting",
+	"webtest", "storageaccountsqueueservice",
+	// excluding dbformysql/user as is not a resource
+	"user",
+}
+
+type SamplesTester struct {
+	tc            *KubePerTestContext
+	decoder       runtime.Decoder
+	UseRandomName bool
+}
+
+type SampleObject struct {
+	SamplesMap   map[string]*linkedhashmap.Map
+	childObjects map[string]genruntime.ARMMetaObject
+}
+
+func NewSampleObject() SampleObject {
+	return SampleObject{
+		SamplesMap:   make(map[string]*linkedhashmap.Map),
+		childObjects: make(map[string]genruntime.ARMMetaObject),
+	}
+}
+
+func NewSamplesTester(tc *KubePerTestContext, useRandomName bool) *SamplesTester {
+	return &SamplesTester{
+		tc:            tc,
+		decoder:       serializer.NewCodecFactory(tc.GetScheme()).UniversalDecoder(),
+		UseRandomName: useRandomName,
+	}
+}
+
+func (t SamplesTester) LoadSamples(rg *resources.ResourceGroup, group string) (*SampleObject, *SampleObject) {
+
+	samples := NewSampleObject()
+	depSamples := NewSampleObject()
+
+	groupPath := getGroupPath(group)
+
+	err := filepath.Walk(groupPath,
+		func(filePath string, info os.FileInfo, err error) error {
+
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() && !isExclusion(filePath) {
+				//if !info.IsDir() && !isExclusion(filePath) {
+
+				sample := t.getObjectFromFile(filePath)
+
+				sample.SetNamespace(t.tc.Namespace)
+
+				if !strings.Contains(filePath, RefsPackage) {
+					if t.UseRandomName {
+						sample.SetName(t.tc.NoSpaceNamer.GenerateName(""))
+					}
+					handleObject(sample, rg, &samples)
+				} else {
+					// We don't change name for dependencies as they have fixed refs which we can't access inside the object
+					// need to make sure we have right names for the refs in samples
+					handleObject(sample, rg, &depSamples)
+				}
+			}
+
+			return nil
+		})
+
+	t.tc.Expect(err).To(BeNil())
+
+	// We process the child objects once we have all the parent objects in the tree already
+	handleChildObjects(t.tc, &samples)
+	handleChildObjects(t.tc, &depSamples)
+
+	return &samples, &depSamples
+}
+
+func (t SamplesTester) getObjectFromFile(path string) genruntime.ARMMetaObject {
+	jsonMap := make(map[string]interface{})
+
+	byteData, err := ioutil.ReadFile(path)
+	t.tc.Expect(err).To(BeNil())
```

Comment:
These `err` checks should be more idiomatic, with the method returning `(genruntime.ARMMetaObject, error)` and the calling code (line #83) using `t.tc.Expect()` to do the check in one place.